pkg/arvo/app/chat-store.hoon

```diff
@@ -34,14 +39,18 @@
   ++  on-load
     |=  old-vase=vase
     =/  old  !<(versioned-state old-vase)
-    ?:  ?=(%2 -.old)
+    ?:  ?=(%3 -.old)
```

Comment:
`+on-load`s get really unwieldy when written with `?:` since they grow quadratically with the number of revisions.  A better pattern is successive `=?`s, like here:

https://github.com/urbit/urbit/blob/4fded00005770a84a53ff77a81ba71353f84b4bd/pkg/arvo/sys/vane/gall.hoon#L1674-L1695

This way you upgrade from previous versions one at a time and only need to think about how to advance from the previous revision to the new one.

Reply:
Good point.